tests/integration/security/authorization_test.go

```diff
@@ -940,37 +940,35 @@ func TestAuthorization_Conditions(t *testing.T) {
 			nsB := apps.Namespace2
 			nsC := apps.Namespace3
 
-			cSet := apps.C.Match(echo.Namespace(nsC.Name()))
-
-			var IPC string
-			for i := 0; i < len(t.Clusters()); i++ {
-				IPC += "\"" + getWorkload(cSet[i], t).Address() + "\","
-			}
-			lengthC := len(IPC)
-			IPC = IPC[:lengthC-1]
-			portC := 8090
-			for i := 0; i < len(t.Clusters()); i++ {
-				t.NewSubTest(fmt.Sprintf("IpA IpB IpC in %s", t.Clusters()[i].StableName())).Run(func(t framework.TestContext) {
-					podAWithIPA := apps.A.Match(echo.InCluster(t.Clusters()[i])).Match(echo.Namespace(nsA.Name()))[0]
-					podBWithIPB := apps.B.Match(echo.InCluster(t.Clusters()[i])).Match(echo.Namespace(nsB.Name()))[0]
-
-					args := map[string]string{
-						"NamespaceA": nsA.Name(),
-						"NamespaceB": nsB.Name(),
-						"NamespaceC": nsC.Name(),
-						"IpA":        getWorkload(podAWithIPA, t).Address(),
-						"IpB":        getWorkload(podBWithIPB, t).Address(),
-						"IpC":        IPC,
-						"PortC":      fmt.Sprintf("%d", portC),
-					}
-
-					policies := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, "testdata/authz/v1beta1-conditions.yaml.tmpl"))
-					t.Config().ApplyYAMLOrFail(t, "", policies...)
+			c := apps.C.Match(echo.Namespace(nsC.Name()))
```

Comment:
I'm a bit confused why the security tests use `echo.Namespace` matcher. Looking at the set of apps, it seems we don't have instances where `apps.{A, B, C, etc.}` contain instances from multiple namespaces.

Reply:
By refactoring tests in authorization_test.go, my understanding is, there are some security policies that allows/denies requests by checking sender's namespace. E.g. a workload `d` allows request sent by `c` from ns1, and denies request sent by `c` from ns2. There are also some security policies that allows/denies request by checking principals. E.g. a workload `c` from ns1 has principal a `cluster.local/ns/ns1/sa/c`, if workload `d` only allows request with this principal, then `d` will deny requests by `c` from other namespaces. There are other tests that verify the security policy works correctly per workload.

Reply:
I missed before that `c` is not unique to a single namespace.

Reply:
https://github.com/istio/istio/pull/32121 – this should make the tests easier to differentiate :)